src/Mantasflowers.WebApi/Controllers/UserController.cs

```diff
@@ -0,0 +1,226 @@
+﻿using FirebaseAdmin.Auth;
+using Mantasflowers.Persistence.Authentication;
+using Mantasflowers.WebApi.Extensions;
+using Mantasflowers.WebApi.Responses;
+using Mantasflowers.WebApi.Setup.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Mantasflowers.WebApi.Controllers
+{
+    [ApiController]
+    [Route("/firebase")]
+    public class UserController : ControllerBase
+    {
+        private readonly ILogger _logger = Log.ForContext<UserController>();
+
+        private readonly FirebaseContext _fbContext;
+
+        private readonly IHttpClientFactory _clientFactory;
+
+        private readonly string _requestUri;
+
+        public UserController(FirebaseContext fbContext, IHttpClientFactory clientFactory, IOptionsMonitor<WebApiKey> optionsAccessor)
+        {
+            _fbContext = fbContext;
+            _clientFactory = clientFactory;
+            _requestUri = $"https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key={optionsAccessor.CurrentValue.Value}";
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="email"></param>
+        /// <param name="password"></param>
+        /// <returns></returns>
+        [HttpGet("get-token")]
```

Comment:
Isn't this "/login"?

Reply:
Too much abstraction can hurt: for example, by adding another endpoint, called "revoke-refresh-token", we can no longer hide the detail that it is about tokens.

Reply:
But even the google endpoint that you end up calling says `signInWithPassword` 🤔

Reply:
I only named this for consistency. Another example, an endpoint, called "refresh-id-token".